packages/node_modules/@ciscospark/redux-module-activity/src/actions.js

```diff
@@ -218,8 +218,10 @@ export function submitActivity(conversation, activity, user, spark) {
  * @returns {function}
  */
 export function setUserTyping(isTyping, conversation, spark) {
-  return (dispatch) =>
-    spark.internal.conversation.updateTypingStatus(conversation.toJS(), {typing: isTyping})
+  return (dispatch, getState) => {
+    const activity = getState().activity;
+    if (activity.getIn([`status`, `isTyping`]) !== isTyping) {
```

Comment:
Since we have this check now in the container, do we want to get rid of the state check in the action and keep `getState` out of it? Up to you.

Reply:
The check here is a bit different than the check in the container. The one in the container doesn't let you set isTyping to true multiple times. This prevents setting to false multiple times as well.